### DISCUSSION TITLE
Rerender Issue in Storybook 7.5.2 with Web Components (Force DOM Rerender)

### DISCUSSION THREAD

**Discussion by ads6495** [storybookjs/storybook#25484] [Help]
### Summary

Hi Storybook team,

I'm encountering a rerender issue while working on an internal design system using Storybook 7.5.2 and web components. The data is logging correctly on user input, but the DOM isn't reflecting the changes.

```
import {html} from 'lit'

import {Meta, StoryObj} from '@storybook/web-components'

export default {
    title: 'Packages/select',
    component: 'wc-select'
} as Meta

export const Select: StoryObj = {
    render: () => {
        const data = [
            {id: 1, name: 'Lorem', value: 'Ipsum'}
            {id: 2, name: 'Dubar', value: 'Dubar'}
            {id: 3, name: 'Bar', value: 'Foo'}
        ]
    let filteredResult: any = data;
    const filterByName= (e) => {
    const {value} = e.target as HTMLInputElement
    filteredResult = data.filter(item => {
        item.name.toLowerCase().inclues(value.toLowerCase())
    })
    }

    return html`
        <wc-form-field>
            <input name="auto" id="autoinput" @input=${filterByName} />
            <wc-autocomplete>  
            ${repeat(data, (item) => html`<wc-option value=${item.name}> ${item.name} </wc-option>`)}
            </wc-autocomplete>
        </wc-form-field>
    `
}
}
```

observations/issues:

- The filteredResult array is being updated as expected when the user types.
- The DOM, however, isn't reflecting the updated filteredResult.

- How can we force a DOM rerender in this scenario to ensure the UI updates match the data changes?
- Are there any known limitations or considerations for rerendering in Storybook 7.5.2 with web components, particularly when using Lit?
- Could this issue be related to the use of repeat within the html template literal? Is there a more recommended approach for handling dynamic content with Lit in Storybook?

Thank you for your time and assistance!

### Additional information

Storybook version: 7.5.2
Relevant packages: @storybook/web-components, lit

**Comment by break-stuff** [comment]
It looks like you are repeating over `data` and not `fileteredResult`. If you update that, do you see a difference?

**Reply by ads6495** [reply]
Hey! thanks for looking into it! Yea I did end up changing it to iterate over filteredResult with no luck.

**Reply by break-stuff** [reply]
The other thing that may be an issue is that `filteredResult` is not reactive, so it never triggers a re-render of the Lit template. You could try using it like you would with vanilla HTML and move that logic to a `<script>` tag.
